services/QuillLMS/app/controllers/ortto_integration/webhooks_controller.rb

```diff
@@ -3,35 +3,27 @@
 module OrttoIntegration
   class WebhooksController < ApplicationController
     protect_from_forgery except: :create
-    before_action :valid_params?, only: [:create]
+    before_action :authenticate
 
     class OrttoWebhookBadRequestException < StandardError; end
 
-    VALID_ACTIONS = [
-      SUBSCRIBE   = 'subscribe',
-      UNSUBSCRIBE = 'unsubscribe'
-    ]
-
     def create
-      unless valid_params?
+      email = params[:email]
```

Comment:
Since it doesn't seem necessary to query User if email is nil, what about adding a guard clause early on?

```
email = params[:email]
return bad_request if email.nil?

user = User.find_by(email: email)
return bad_request if user.nil?
...

private def bad_request
  ErrorNotifier.report(OrttoWebhookBadRequestException.new("Bad request with params: #{params}"))
  head :accepted
end
```